### PR TITLE
Image Block: Adding an image upload button

### LINKED ISSUE #1702: Add "upload" button to image block placeholder
Right now we have a media library button, but it'd be good to also have a button to upload a new item right there.

### PR DESCRIPTION
**Pull request by youknowriad**
to upload an image without opening the media modal.

closes #1702

### CODE REVIEW

File: components/form-file-button/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
```
Comment (mtias) [inline components/form-file-button/index.js]: Should we call this `FileUpload`? I don't think I would have found it easily with this name.

Comment (youknowriad) [inline components/form-file-button/index.js]: I try to group all the "form" related components by calling them `Form*` but I don't mind changing.

Comment (mtias) [inline components/form-file-button/index.js]: `FormFileUpload` maybe

**Comment by mtias** [comment]
Works great for me!

**Comment by mtias** [comment]
@jasmussen the upload button has a hover effect that the "insert from media library" does not. Not sure what the status should be.

**Comment by jasmussen** [comment]
> the upload button has a hover effect that the "insert from media library" does not. Not sure what the status should be.

The blue hover effect is caused by the button having a `components-icon-button` CSS class. We can either address the CSS or remove the class. What makes the most sense?

**Comment by youknowriad** [comment]
Fixed the hover color, kept the class because it's automatically included in the IconButton component.

**Comment by mkaz** [comment]
Looks good, it works for me.

I want to use the same method for the Gallery block and uploading of multiple images. Looking at the code, it looks like you have support for multiple in the `form-file-upload` button, however the function that processes the uploads is in the image block, could we break that out to be independent, so any block can use it.

:+1: to commit and I can abstract it out for you if you like

**Comment by youknowriad** [comment]
@mkaz yeah, we'd probably want to extract this function somewhere. Let's merge this and will let you abstract while working on the gallery.

Thanks everyone for the reviews. Merging

**Pull request merged**